packages/api-server/src/db/helpers.ts

```diff
@@ -289,3 +304,44 @@ export function buildQueryLogAddress(
     );
   }
 }
+
+/*
+return a slice of log array which satisfy the topics matching.
+matching rule:
+      Topics are order-dependent.
+	Each topic can also be an array of DATA with “or” options.
+	[example]:
+
+	  A transaction with a log with topics [A, B],
+	  will be matched by the following topic filters:
+	    1. [] “anything”
+	    2. [A] “A in first position (and anything after)”
+	    3. [null, B] “anything in first position AND B in second position (and anything after)”
+	    4. [A, B] “A in first position AND B in second position (and anything after)”
+	    5. [[A, B], [A, B]] “(A OR B) in first position AND (A OR B) in second position (and anything after)”
+
+	source: https://eth.wiki/json-rpc/API#eth_newFilter
+*/
+export function buildQueryLogTopics(
+  queryBuilder: KnexType.QueryBuilder,
+  topics: FilterTopic[]
+) {
+  if (topics.length !== 0) {
+    queryBuilder.whereRaw(`array_length(topics, 1) >= ${topics.length}`);
```

Comment:
```suggestion
    queryBuilder.whereRaw("array_length(topics, 1) >= ?", topics.length);
```

Please use [`Raw Parameter Binding`](http://knexjs.org/guide/raw.html#raw-parameter-binding) to avoid sql injection